connector/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -194,6 +194,10 @@ private[kafka010] class KafkaMicroBatchStream(
     val startPartitionOffsets = start.asInstanceOf[KafkaSourceOffset].partitionToOffsets
     val endPartitionOffsets = end.asInstanceOf[KafkaSourceOffset].partitionToOffsets
 
+    if (allDataForTriggerAvailableNow != null) {
```

Comment:
This is effectively done via two assertions (topic partitions, offsets) in assertEndOffsetForTriggerAvailableNow.

Do you mean by having sanity check for non Trigger.AvailableNow case? KafkaOffsetReader.getOffsetRangesFromResolvedOffsets handles it, and we have to lean on this because setting failOnDataLoss to false allows end topic offsets is smaller than start topic offsets.

(I would argue it is scary how many fault scenarios the option simply tolerates (ignores) but it's not something we can make changes.)